src/app/modals/coldstake/coldstake.component.ts

```diff
@@ -0,0 +1,184 @@
+import { Component, OnInit, Inject, forwardRef } from '@angular/core';
+import { Log } from 'ng2-logger';
+
+import { ModalsService } from '../modals.service';
+import { RPCService } from '../../core/rpc/rpc.module';
+
+import { PasswordComponent } from '../shared/password/password.component';
+
+@Component({
+  selector: 'app-coldstake',
+  templateUrl: './coldstake.component.html',
+  styleUrls: ['./coldstake.component.scss']
+})
+export class ColdstakeComponent implements OnInit {
+
+  log: any = Log.create('coldstake.component');
+
+  /*  State  */
+  step: number = 0;
+  type: string;
+
+
+  /*  Hot wallet (step 3)  */
+
+  hotStakeAddress: any = 'Generating...';
+
+  /*  Cold wallet (step 3)  */
+  prevColdStakeAddress: any = '';
+  coldStakeAddress: any = '';
+  private validAddress: boolean = undefined;
+  /*  Cold wallet (step 4)  */
+  finalMessage: string = '';
+
+  constructor(
+    @Inject(forwardRef(() => ModalsService))
+    private _modalsService: ModalsService,
+    private _rpc: RPCService
+  ) { }
+
+  ngOnInit() {
+  }
+
+  nextStep () {
+    this.log.d(`Going to step: ${this.step + 1}`)
+    this.step++;
+  }
+
+  create(type: string) {
+    if (['hot', 'cold'].indexOf(type) !== -1) {
+      this.type = type;
+    }
+
+
+    if (['Locked', 'Unlocked, staking only'].indexOf(this._rpc.state.get('encryptionstatus')) === -1) {
+      // unlock wallet and send transaction
+      this.nextStep();
+    }
+
+    if (type === 'hot') {
+      this.rpc_retrieveHotWallet();
+    }
+
+  }
+
+
+  /** called when the hot wallet unlocked (by password component)  */
+  unlockHotWallet(encryptionStatus: String) {
+    if (this.step === 1) {
+      this.nextStep();
+    }
+
+    if (encryptionStatus === 'Unlocked') {
+      this.rpc_retrieveHotWallet();
+    } else {
+      this.log.d(`unlockHotWallet, did not unlock: ${encryptionStatus}`);
+    }
+  }
+
+
+  rpc_retrieveHotWallet(): void {
+    this.log.d('rpc_retrieveHotWallet called');
+    this._rpc.call('extkey', [
+      'account'
+    ])
+    .subscribe(
+      success => {
+        this.rpc_filterHotListOfAccounts(success);
+      },
+      error => {
+        this.log.i('rpc_retrieveHotWallet: ', error);
+      });
+  }
+
+  rpc_filterHotListOfAccounts(response: Object) {
+    const chains: Array<Object> = response['chains'];
+
+    // get our cold stake chain if it exists
+    const hotStakeChain: Array<Object> = chains.filter(function(chain: Object){
+      return chain['label'] === 'Cold Staking';
```

Comment:
use chain.label, set type to any, otherwise we're using TypeScript for no reason.. We need to create interfaces at a later stage, so we should code for it.